### DISCUSSION TITLE
Workflow Reuse error: missing a `on.workflow_call` trigger

### DISCUSSION THREAD

**Discussion by jlbooker** [community/community#38954] [Actions] [answered]
### Select Topic Area

Bug

### Body

Following the instructions at https://docs.github.com/en/actions/using-workflows/reusing-workflows, I added an `on.workflow_call:` line to my workflow in order to reuse it. When calling the workflow, I keep getting the error:
`error parsing called workflow "{my_org}/{my_repo}/.github/workflows/node.js.yml@main": workflow is not reusable as it is missing a `on.workflow_call` trigger`

I even copied the full reusable workflow example into the file and I'm still getting the same error. What am I doing wrong here?

**Comment by airtower-luna** [comment]
What *exactly* does the trigger configuration look like? A subtle syntax error seems the most likely explanation to me. :sweat_smile:

**Reply by jlbooker** [reply]
node.js.yml:
```
name: Reusable workflow example

on:
  workflow_call:
    inputs:
      config-path:
        required: true
        type: string
    secrets:
      token:
        required: true

jobs:
  triage:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/labeler@v4
        with:
          repo-token: ${{ secrets.token }}
          configuration-path: ${{ inputs.config-path }}
```

Called by release.yml:
```
name: Release

on:
  push:
    # Create a release when a new version tag is pushed to main
    branches: ['main']
    tags: ['v*'] # Push events to matching v*, i.e. v1.0, v20.15.10

jobs:
  build:
    uses: [org]/[repo]/.github/workflows/node.js.yml@main
```

Full error:
```
Invalid workflow file: .github/workflows/release.yml#L11(https://github.com/[org]/]repo]/actions/runs/[runid]/workflow)
error parsing called workflow "[org]/[repo]/.github/workflows/node.js.yml@main": workflow is not reusable as it is missing a `on.workflow_call` trigger
```

Clearly node.js.yml _has_ an `on.workflow_call` trigger, so I don't understand what I'm missing.

**Reply by airtower-luna** [reply]
Does the version of `node.js.yml` on the `main` branch contain the trigger?
```yaml
    uses: [org]/[repo]/.github/workflows/node.js.yml@main
```

**Reply by jlbooker** [reply]
Yes, the version above is what I have committed to the main branch. It was originally and node JS build, but I replaced it with the "reusable workflow example" content from the tutorial, just to see if I could get the example working.

**Reply by airtower-luna** [reply]
Then I'm out of ideas as long as you can't share a link to the actual workflow log, I'm afraid. A subtle issue somewhere still seems most likely. :sweat_smile: 

If you're sure it's not that, you could also try to [send a bug report to support](https://support.github.com/contact/bug-report).

**Reply by jlbooker** [reply]
Thanks for the help! Here's the latest failed run:
https://github.com/BrownBook/ExperientialLearningInventory/actions/runs/3453021132

Here's the workflows directory on the main branch: https://github.com/BrownBook/ExperientialLearningInventory/blob/main/.github/workflows/

I'm trying to call node.js.yml from release.yml. I've updated node.js.yml back to my version, instead of the tutorial example, but you can see the history that's also failed.

**Reply by ChristopherHX** [reply]
Wow, your reusable workflow is 100% correct if you copy it to another repository or use the sha as version
If I directly call your workflow from your repo and branch it fails for me too.

This works
```yaml
name: Release

on:
  push:

jobs:
  build:
    uses: BrownBook/ExperientialLearningInventory/.github/workflows/node.js.yml@a82d8929306d9bb8c235164b8134bff39f70d5f3
```

An old version of your workflow must be cached by GitHub somewhere.

**Reply by jlbooker** [reply]
Thanks for confirming and giving the workaround using the @sha notation. I've filed a bug report. :-)

**Comment by jlbooker** [comment, marked as answer]
I believe I have discovered the cause of this problem. I had managed to create (I believe through a different Action) a tag named `main`, in addition to the branch named `main`. Instead of using the version from the `main` branch, I believe the Workflow was continuing to use version of `node.js.yml` from the `main` _tag_, which was missing the `on.workflow_call` trigger as correctly indicated by the error message.

In short, the error message was accurate, but it wasn't clear that the Workflow being used was from the `main` _tag_ and not the `main` _branch_.

**Reply by Kystabell3410** [reply]
Can you help me with the same problem

**Reply by Kystabell3410** [reply]
Because I have been trying to find the best way to know what was happening with it

**Reply by getellez** [reply]
How can we point it to use the branch name instead of the tag?
